coldfront/plugins/qumulo/tests/test_tasks.py

```diff
@@ -245,3 +260,284 @@ def test_conditionally_update_storage_allocation_statuses_checks_all_pending_all
+
+        self.quotas = self.__get_qumulo_quota_data(
+            get_mock_quota_data(self.STORAGE2_PATH)
+        )
+
+        self.status_active = AllocationStatusChoiceFactory(name="Active")
+        self.status_ready_for_deletion = AllocationStatusChoiceFactory(
+            name="Ready for Deletion"
+        )
+
+        for index, (path, value) in enumerate(
+            get_mock_quota_data(self.STORAGE2_PATH).items()
+        ):
+            if "exclude" in path:
+                continue
+
+            form_data = {
+                "storage_filesystem_path": path.rstrip("/"),
+                "storage_export_path": path.rstrip("/"),
+                "storage_name": f"for_tester_{index}",
+                "storage_quota": value["limit"],
+                "protocols": ["nfs"],
+                "rw_users": [f"user_{index}_rw"],
+                "ro_users": [f"user_{index}_ro"],
+                "storage_ticket": f"ITSD-{index}",
+                "cost_center": "Uncle Pennybags",
+                "department_number": "Time Travel Services",
+                "service_rate": "general",
+            }
+
+            # create_allocation(project=self.project, user=self.user, form_data=form_data)
+            allocation = create_allocation(
+                project=self.project, user=self.user, form_data=form_data
+            )
+            allocation.status = self.status_active
+            allocation.save()
+
+        self.storage_filesystem_path_attribute_type = (
+            AllocationAttributeType.objects.get(name="storage_filesystem_path")
+        )
+        self.storage_quota_attribute_type = AllocationAttributeType.objects.get(
+            name="storage_quota"
+        )
+
+        return super().setUp()
+
+    def tearDown(self) -> None:
+        os.environ["STORAGE2_PATH"] = self.original_storage_path
+
+        return super().tearDown()
+
+    def __get_qumulo_quota_data(self, quota_data: dict) -> dict:
+        quotas = list(
+            map(
+                lambda quota_key_value: (
+                    {
+                        "id": quota_key_value[1]["id"],
+                        "path": quota_key_value[0],
+                        "limit": quota_key_value[1]["limit"],
+                        "capacity_usage": quota_key_value[1]["usage"],
+                    }
+                ),
+                quota_data.items(),
+            )
+        )
+        return {"quotas": quotas, "paging": {"next": ""}}
+
+    def test_qumulo_result_set_page_limit_should_be_set(self) -> None:
+        page_limit = qumulo_api.QumuloAPI.get_result_set_page_limit()
+        self.assertIsNotNone(page_limit)
+
+    @skip("Until we have a chance to propagte the ENV variable.")
+    def test_qumulo_result_set_page_limit_should_raise_an_exception_if_not_set(
+        self,
+    ) -> None:
+        old_page_limit = os.environ.get("QUMULO_RESULT_SET_PAGE_LIMIT")
+        os.environ["QUMULO_RESULT_SET_PAGE_LIMIT"] = ""
```

Comment:
@harterjwustl I am curious to know why not using mock?

Reply:
see: "I believe (and my testing shows) this only mocks it locally. Since the mock data is being being build elsewhere, it isn't getting the same data."  I don't believe the target method will be see the patched dict